pkg/storage/pebble.go

```diff
@@ -111,6 +111,18 @@ var columnarBlocksEnabled = settings.RegisterBoolSetting(
 	settings.WithPublic,
 )
 
+// deleteCompactionsCanExcise controls whether delete compactions can
+// apply rangedels/rangekeydels on sstables they partially apply to, through
+// an excise operation, instead of just applying the rangedels/rangekeydels
+// that fully delete sstables.
+var deleteCompactionsCanExcise = settings.RegisterBoolSetting(
+	settings.SystemOnly,
+	"storage.delete_compaction_excise.enabled",
+	"set to false to direct Pebble to not partially excise sstables in delete-only compactions",
+	metamorphic.ConstantWithTestBool(
+		"storage.delete_compaction_excise.enabled", true), /* defaultValue */
+	settings.WithPublic)
```

Comment:
Driveby: should this be a public setting?

Reply:
I think there's a good argument for it to go either way. For now I just copied the same visibility as some of the other compaction-y settings like ingest_split.enabled above.a